Team-4-submission

waynewangyuxuan opened the pull request:
Merged Commits into 2

digit-google committed:
Add InputsCollector class.

Add a new class to wrap the logic needed to implement the
`inputs` tool correctly (see Issue #2482 for details), and
provide a unit-test for it.

digit-google committed:
Fix `inputs` tool implementation.

This uses the InputsCollector class introduced in the
previous patch to implement the tool properly. Results
are still shell-escaped and sorted alphabetically.

Fixed #2482

digit-google committed:
Add new options to `inputs` tool.

Add new options to the `inputs` tool in order to change
the format of its output:

- `--no-shell-escape` to avoid shell-escaping the results.

- `--dependency-order` to return results in dependency order,
  instead of sorting them alphabetically.

- `--print0` to use \0 as the list separator in the list,
  useful to process the target paths with `xargs -0` and
  similar tools.

YuWei-CH committed:
init

CircleCI Commit

Update CI

update README.md

add sudo

Updated CMakeLists.txt to include clang-tidy configuration

change clang test name

update clangtest file

update clang-tidy/circleCI use. Learned from team5's implementation

update yml file

typos

init

CircleCI Commit

Update CI

update README.md

add sudo

test folder wrong

kyotov commented:
do you have a CI link i can look at?

waynewangyuxuan commented:
https://app.circleci.com/pipelines/circleci/XuhSKKLtTmft9W3VV6Pa6t/Pht6yzSCo6GJvUbVC1gNV8
Hi professor, this is the link.

On Tue, Oct 8, 2024 at 9:42 PM Kamen Yotov ***@***.***> wrote:

> do you have a CI link i can look at?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nyuoss/ninja-shadowdash-nyu/pull/8#issuecomment-2401105908>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYEIBD257F3ZB5GC66ZO46TZ2SCZFAVCNFSM6AAAAABPRDIMGSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBRGEYDKOJQHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kyotov commented:
seems like you can do a bit more to integrate this part?
<img width="754" alt="image" src="https://github.com/user-attachments/assets/e4960487-f149-484f-9a27-b447d4c4714c">

waynewangyuxuan commented:
We overlooked this part before…
I think we can.
Btw, the ci/cd on master, the (most recent ) failed one, is what we intend
to show you. The ci/cd did not past because we did not pass the clang-tidy
check. We put the clang-tidy into crest, wondering whether we should do it
at build or do it with ctest like it is?


On Tue, Oct 8, 2024 at 9:51 PM Kamen Yotov ***@***.***> wrote:

> seems like you can do a bit more to integrate this part?
> image.png (view on web)
> <https://github.com/user-attachments/assets/e4960487-f149-484f-9a27-b447d4c4714c>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nyuoss/ninja-shadowdash-nyu/pull/8#issuecomment-2401112881>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYEIBD3FO4A3QEF2PGIDTZ3Z2SDYJAVCNFSM6AAAAABPRDIMGSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMBRGEYTEOBYGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

kyotov commented on .circleci/config.yml:
```diff
@@ -0,0 +1,35 @@
+jobs:
+  build-test:
+    docker:
+      - image: cimg/base:current
```
for extra credit you can build your own image, but for this hw, this is ok

kyotov commented on .circleci/config.yml:
```diff
@@ -0,0 +1,35 @@
+            echo "Installing clang-tidy"
+            sudo apt-get update && sudo apt-get install -y clang-tidy
+            echo "Installed"
+      - run:
+          name: Build
+          command: | 
+            echo "Building"
+            mkdir build && cd build
+            cmake -S ..
+            make -j $(nproc)
+            cd ../
+            echo "Built"
+      - run:
+          name: Test
+          command: | 
+            echo "Start Testing"
+            cd build
+            ctest
+            echo "Testing Done"
```
where is the clang-tidy running?

kyotov commented on CMakeLists.txt:
```diff
@@ -312,6 +312,45 @@ if(BUILD_TESTING)
   add_test(NAME NinjaTest COMMAND ninja_test)
+
+
+  set(NINJA_TEST_SOURCES
+    src/build_log_test.cc
+    src/build_test.cc
+    src/clean_test.cc
+    src/clparser_test.cc
+    src/depfile_parser_test.cc
+    src/deps_log_test.cc
+    src/disk_interface_test.cc
+    src/dyndep_parser_test.cc
+    src/edit_distance_test.cc
+    src/elide_middle_test.cc
+    src/explanations_test.cc
+    src/graph_test.cc
+    src/json_test.cc
+    src/lexer_test.cc
+    src/manifest_parser_test.cc
+    src/missing_deps_test.cc
+    src/ninja_test.cc
+    src/state_test.cc
+    src/string_piece_util_test.cc
+    src/subprocess_test.cc
+    src/test.cc
+    src/util_test.cc
+	)
```
this looks like something that should be in the original ninja?
what is the purpose?

waynewangyuxuan replied:
Oh, we will delete this in here, we did not write any code at the time, so we were checking the src code see whether clang-tidy was working

kyotov commented on CMakeLists.txt:
```diff
@@ -312,6 +312,45 @@ if(BUILD_TESTING)
+  find_program(CLANG_TIDY_EXE NAMES "clang-tidy" REQUIRED)
+  # Apply clang-tidy checks to the target
+# Add a custom test for clang-tidy
+   if(CLANG_TIDY_EXE)
+    add_test(
+        NAME clang-tidy-check
+        COMMAND ${CLANG_TIDY_EXE} ${NINJA_TEST_SOURCES} --checks=-*,modernize-* --warnings-as-errors=*
+    )
+    else()
+    	message(WARNING "clang-tidy not found, skipping clang-tidy test")
+    endif()
```
i see that you run it now, but this is not the normal way to run clang-tidy in cmake (i think)...
can you check?

waynewangyuxuan replied:
I think you are right about this, we will update this

kyotov reviewed (changes requested):
please address the comments and let me know when it is ready for another look

waynewangyuxuan commented:
> please address the comments and let me know when it is ready for another look

Hi, professor, the use of clang-tidy is fixed

kyotov commented on .circleci/config.yml:
```diff
@@ -1,35 +1,43 @@
-
+          name: Run Tests
+          command: |
+            ./cicd-build/ninja_test
```
use ctest
add perf tests

kyotov commented on steps/hello-world-graph/libninja.dylib:
this should be in gitignore

waynewangyuxuan committed:
add test result display for circleCI dashboard

Added ctest and performance test.

Try to show test on CI/CD dashboard, followed the instruction, however
we could not get it working somehow.
